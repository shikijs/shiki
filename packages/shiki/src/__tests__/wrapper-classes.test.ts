import { getHighlighter } from '../index'

test('wrapperClasses parameter adds classes', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord'
  })

  const out = highlighter.codeToHtml(`console.log('test');`, 'js', 'nord', ['foo', 'bar'])
  expect(out).toMatchSnapshot()
  expect(out).not.toContain('shiki')
})

test('empty wrapperClasses omits <pre> class attribute', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord'
  })
  const out = highlighter.codeToHtml(`console.log('test');`, 'js', 'nord', [])
  expect(out).toMatchSnapshot()
  expect(out).not.toContain('<pre class=')
})
