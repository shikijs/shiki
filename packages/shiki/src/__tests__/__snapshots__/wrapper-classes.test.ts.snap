// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty wrapperClasses omits <pre> class attribute 1`] = `"<pre style=\\"background-color: #2e3440ff\\"><code><span class=\\"line\\"><span style=\\"color: #D8DEE9\\">console</span><span style=\\"color: #ECEFF4\\">.</span><span style=\\"color: #88C0D0\\">log</span><span style=\\"color: #D8DEE9FF\\">(</span><span style=\\"color: #ECEFF4\\">&#39;</span><span style=\\"color: #A3BE8C\\">test</span><span style=\\"color: #ECEFF4\\">&#39;</span><span style=\\"color: #D8DEE9FF\\">)</span><span style=\\"color: #81A1C1\\">;</span></span></code></pre>"`;

exports[`wrapperClasses parameter adds classes 1`] = `"<pre class=\\"foo bar\\" style=\\"background-color: #2e3440ff\\"><code><span class=\\"line\\"><span style=\\"color: #D8DEE9\\">console</span><span style=\\"color: #ECEFF4\\">.</span><span style=\\"color: #88C0D0\\">log</span><span style=\\"color: #D8DEE9FF\\">(</span><span style=\\"color: #ECEFF4\\">&#39;</span><span style=\\"color: #A3BE8C\\">test</span><span style=\\"color: #ECEFF4\\">&#39;</span><span style=\\"color: #D8DEE9FF\\">)</span><span style=\\"color: #81A1C1\\">;</span></span></code></pre>"`;
